snet_cli/mpe_service_metadata.py

```diff
@@ -171,6 +171,14 @@ def get_payment_address(self, group_name = None):
     def get_all_endpoints(self):
         return [e["endpoint"] for e in self.m["endpoints"]]
 
+    def get_all_endpoints_with_group_name(self):
+        endpts_with_grp = {}
+        for e in self.m["endpoints"]:
+            if e['group_name'] not in endpts_with_grp.keys():
```

Comment:
you don't need ```.keys()``` here ```x in dict``` is equal to ```x in dict.keys()```  (but I would use defaultdict here to simplify the code, as I've said in my first comment)